managers/models.py

```diff
@@ -8,8 +11,34 @@ class Project(models.Model):
     start_date = models.DateField()
     stop_date = models.DateField(null=True, blank=True)
     terminated = models.BooleanField(default=False)
-    managers = models.ManyToManyField(CustomUser, related_name="managers")
-    members = models.ManyToManyField(CustomUser, related_name="members")
+    managers = models.ManyToManyField(CustomUser, related_name="project_managers")
+    members = models.ManyToManyField(CustomUser, related_name="project_members")
 
     def __str__(self):
         return self.name
+
+
+@receiver(m2m_changed, sender=Project.managers.through)
+def update_user_type(sender, action, pk_set, **kwargs):
+    project = kwargs['instance']
+    if action == 'pre_remove' or action == 'post_remove':
+        remove_manager_from_project_action(pk_set)
+    elif action == 'pre_add' or action == "post_add":
```

Comment:
I think that elif is necessary, because there is one more action post and pre_clear which is using with clear() method. When it would work with save() method from elif I have got recursion error.